### PR TITLE
Update nsxtreceiver to checkapi standards

### PR DESCRIPTION
**Pull request by greatestusername**
**Description:** <Describe what has changed.>
stop exporting `NewMockClient` method to align with checkapi standards as related to issue #26304  

**Link to tracking Issue:** #26304 

**Testing:** 
go run cmd/checkapi/main.go .
go test for nsxtreceiver

**Documentation:** no changes

### CODE REVIEW

File: .chloggen/26304-checkapi-nsxtreceiver.yaml
Comment (atoulme) [inline .chloggen/26304-checkapi-nsxtreceiver.yaml:4]: since this is test code, it's not considered a breaking change, and you can delete the changelog entry. I'll mark this PR as "Skip Changelog". Changes look good otherwise.

Comment (greatestusername) [reply .chloggen/26304-checkapi-nsxtreceiver.yaml:4]: Ah good point! Thank you! Will do

Comment (greatestusername) [reply .chloggen/26304-checkapi-nsxtreceiver.yaml:4]: is this `enhancement` or `bug_fix` or does it even matter?

Comment (songy23) [reply .chloggen/26304-checkapi-nsxtreceiver.yaml:4]: You can simply remove this file since it's a test-only change.

Comment (greatestusername) [reply .chloggen/26304-checkapi-nsxtreceiver.yaml:4]: oh gotcha. Yeah that was in the sentence sure enough... Some days... :brain:

**Review by atoulme — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Review by schmikei — APPROVED** [review]
I can't merge this, but as a codeowner I'm approving as a safe change

**Review by codeboten — APPROVED** [review]

**Pull request merged**
